Bookmarks

genusistimelord opened the pull request:
Added in bookmarks with colors, formatting and children.  The only issue is the bookmark can only point to the ObjectID of a Page. 

I added in the ability to Add in bookmarks to a Document and then phrase thru to outline and add them at the end of the PDF where they should be added. I did not add an ability to remove or sort thru bookmarks.

J-F-Liu commented:
I think `BookMark ` should be renamed to `Bookmark`, and there are some clippy warnings to fix.
Thanks.

genusistimelord commented:
> 
> 
> I think `BookMark ` should be renamed to `Bookmark`, and there are some clippy warnings to fix.
> Thanks.

I have changed BookMark to Bookmark and fixed the clippy messages that I saw. please let me know if i missed any.
Thank you J-F-Liu
